model/change_point_detection.go

```diff
@@ -428,3 +434,42 @@ func createChangePoint(ctx context.Context, env cedar.Environment, resultToUpdat
 	_, err := env.GetDB().Collection(perfResultCollection).UpdateOne(ctx, filter, update)
 	return errors.Wrap(err, "Unable to create change point")
 }
+
+func TriageChangePoints(ctx context.Context, env cedar.Environment, changePoints map[string]string, status TriageStatus) error {
+	session, err := env.GetDB().Client().StartSession()
+	if err != nil {
+		return errors.Wrap(err, "Unable to start session for triage update")
+	}
+	defer session.EndSession(ctx)
+	if err := session.StartTransaction(); err != nil {
+		return errors.Wrap(err, "Unable to start transaction for triage update")
+	}
+	return mongo.WithSession(ctx, session, func(sc mongo.SessionContext) error {
+		for perfResultId, measurement := range changePoints {
+			filter := bson.M{
+				perfIDKey: perfResultId,
+				bsonutil.GetDottedKeyName(perfAnalysisKey, perfAnalysisChangePointsKey, perfChangePointMeasurementKey): measurement,
+			}
+			update := bson.M{
+				"$set": bson.M{
+					bsonutil.GetDottedKeyName(perfAnalysisKey, perfAnalysisChangePointsKey, "$", perfChangePointTriageKey, perfTriageInfoStatusKey):    status,
+					bsonutil.GetDottedKeyName(perfAnalysisKey, perfAnalysisChangePointsKey, "$", perfChangePointTriageKey, perfTriageInfoTriagedOnKey): time.Now(),
+				},
+			}
+			res, err := env.GetDB().Collection(perfResultCollection).UpdateOne(sc, filter, update)
+			fmt.Print(res)
+			if err != nil || res.ModifiedCount != 1 {
+				err2 := session.AbortTransaction(ctx)
+				if err2 != nil {
+					return errors.Wrap(err, "Failed to abort transaction during failed change point triage")
+				}
+				return errors.New(fmt.Sprintf("Unable to triage change point <%s> for performance result %s", measurement, perfResultId))
+			}
+		}
+		err := session.CommitTransaction(sc)
```

Comment:
do we need to retry this? is there special retry logic?

Reply:
My take on this is we'd be retrying due to connection issues --> and that's something the driver itself should manage, right? Otherwise if the transaction simply fails to commit, there is not much we can do, or?

Reply:
I mean it could fail to commit because of some kind of write conflict, and those aren't automatically retried. I'm also not sure that the driver actually retries due to connection issues either

Reply:
The driver doesn't automatically retry, but it should bubble up transient errors such as there not being an available write ticket within the transaction timeout. I'd expect that cedar's db to not hit this at the moment, but it's worth retrying so that it doesn't become a problem in the future

Reply:
Should we wrap the driver in a client then? It seems like this retry logic should be implemented wherever driver/client is consumed, rather than separately all over the place -- right?

Reply:
On a separate not; it sounds like this improvement would apply to most of the codebase and could be done in a separate tix?